storage2/src/snapshot.rs

```diff
@@ -9,38 +9,38 @@ use tracing::Span;
 
 use crate::state::StateRead;
 
-/// Snapshots maintain a point-in-time view of the underlying storage, suitable
-/// for read-only access by multiple threads, i.e. RPC calls.
+mod rocks_wrapper;
+use rocks_wrapper::RocksDbSnapshot;
+
+/// A snapshot of the underlying storage at a specific state version, suitable
+/// for read-only access by multiple threads, e.g., RPC calls.
 ///
-/// This is implemented as a wrapper around a [RocksDB snapshot](https://github.com/facebook/rocksdb/wiki/Snapshot)
-/// with an associated JMT version number for the snapshot.
+/// Snapshots are cheap to create and clone.  Internally, they're implemented as
+/// a wrapper around a [RocksDB
+/// snapshot](https://github.com/facebook/rocksdb/wiki/Snapshot) with a pinned
+/// JMT version number for the snapshot.
 #[derive(Clone)]
-pub(crate) struct Snapshot(pub(crate) Inner);
+pub struct Snapshot(Arc<Inner>);
```

Comment:
This commit simplifies visibility from `pub(crate) Struct(pub(crate) Inner)` to `pub Struct(Inner)`, so that visibility of the outer struct is controlled by whether or not the parent module re-exports it, and the inner type is completely encapsulated.  I think this is generally a bit cleaner.